src/controllers/controller.bar.js

```diff
@@ -383,12 +395,14 @@ module.exports = function(Chart) {
 			var meta = me.getMeta();
 			var scale = me.getValueScale();
 			var datasets = chart.data.datasets;
-			var value = scale.getRightValue(datasets[datasetIndex].data[index]);
+			// float-bar support, if y arguments are array function will use top - bottom value to calculate bar height
+			var yValue = datasets[datasetIndex].data[index];
+			var value = helpers.isArray(yValue) ? (scale.getRightValue(yValue[1]) - scale.getRightValue(yValue[0])) : scale.getRightValue(yValue);
```

Comment:
Based on my comment about `getRightValue`:

```javascript
var rvalue = scale.getRightValue(datasets[datasetIndex].data[index]);
var value = helpers.isArray(rvalue) ? rvalue[1] - rvalue[0] : rvalue;
// ...
var start = helpers.isArray(rvalue) ? rvalue[0] : 0;
```

Reply:
Not sure it will work without further upgrading of getRightValue() function. Currently, this function returns only one value and is not supposed to return array of the values, so

`rvalue[1] - rvalue[0] `

this one will not work currently. I can upgrade this function to return arrays of values if the incoming argument is array. SHould i do this?

Reply:
I would suggest to not upgrade and stick to current functioning, itts the most safe method to keep other partts of the code untouched and not influenced with chages from there.

Reply:
@simonbrunel what do you think about introducing new methods `getMinValue(rawValue)` and `getMaxValue(rawValue)` alongside `getRightValue(rawValue)`? That would allow us to keep the logic of extracting the high and low values from the data point in a single place. This has a lot of benefits. E.g. in the financial chart I don't have to reimplement every single method to have an understanding of `h` and `l`, but can just override those two methods (users pass in data like `{o: 3.34, h: 4.42, l: 3.10, c: 3.93}`)